internal/provider/resource_gitlab_project_approval_rule.go

```diff
@@ -72,6 +77,7 @@ func resourceGitlabProjectApprovalRuleCreate(ctx context.Context, d *schema.Reso
 	options := gitlab.CreateProjectLevelRuleOptions{
 		Name:               gitlab.String(d.Get("name").(string)),
 		ApprovalsRequired:  gitlab.Int(d.Get("approvals_required").(int)),
+		RuleType:           gitlab.String(d.Get("rule_type").(string)),
```

Comment:
If an attribute is not set in the config and is optional it's good to also not send it to the API, therefore, I'd do it like this:

https://github.com/gitlabhq/terraform-provider-gitlab/blob/278f2b394ec57caf9e415c2b42d31bdb48b76e78/internal/provider/resource_gitlab_topic.go#L56

I'm aware that it also applies to the attributes already in here. It's never to late to start though and if you want feel free to apply it for the others, too 🌮

Reply:
I would prefer to touch only this one in this scope and then go through the others separately, would that be ok?

Reply:
sure thing! You can also leave it and I'll sweep over it afterwards ;)